lib/propagation/w3c.js

```
@@ -59,6 +62,11 @@ function marshalTraceContext(context) {
 function marshalTraceContextv1(context) {
   const spanId = util.currentSpanId(context);
 
+  // do not propagate non-standard ids
+  if (!context.id.match(TRACE_ID_REGEX) || !spanId.match(SPAN_ID_REGEX)) {
+    return;
```

hm, this doesn't an indication as to why it was rejected.

I'd imagine in golang if the return type is `string`, then "" would be returned in that code.  I think that'd be a good change here (and would make api docs easier to deal with) - undefined and "" are both falsey, but the latter means that folks don't have to actually _check_ if they don't want to.  anywhere they'd put a result from `marshalTraceContext`, they can still put it.

if we actually do want to do error handling there (in golang to continue the ... analogy), we would probably change the return type to `(string, error)` with these being:

1. `return nil, errors.New("illegal w3c trace id")`
2. `return nil, errors.New("illegal w3c span id")`

respectively.  That means instead of returning here we should _probably_ throw an exception.

I don't know what the error handling story is for the other beelines.  Honestly throwing an exception feels like the right thing for JS, but until we've decided more widely, "" feels like the right thing to do.

---

Agreed. Changing to empty string and adding some debug messaging as well